src/index.ts

```diff
@@ -1,36 +1,52 @@
-const { promisify } = require('promise-callbacks');
+import { promisify } from 'promise-callbacks';
 
-async function promiseIterate(iterable, iteratee, { batchSize } = {}) {
+type Iterable<T> = {
+  next: (p?: T) => Promise<T | void>;
+};
+type Iteratee<T> = (p: T) => Promise<T>;
+
+async function promiseIterate(
+  iterable: Iterable<any>,
+  iteratee: Iteratee<any>,
+  { batchSize }: { batchSize?: number } = {}
+): Promise<void> {
```

Comment:
Does it make sense to use generics here too?

```suggestion
async function promiseIterate<T>(
  iterable: Iterable<T>,
  iteratee: Iteratee<T>,
  { batchSize }: { batchSize?: number } = {}
): Promise<void> {
```

And also for `asyncIterate` function below.

Again I'm not sure on that, can we release an alpha version and check with the actual usage of these functions that they don't break?

Reply:
while adding the `Iterable<T>` yesterday I played around with such an approach but it would clutter the consumers of such library, and in the end we would be having something like the following:
<img src="https://github.com/user-attachments/assets/dc02f5a5-1094-435a-b490-a34d952fe1b9" wodth="350" />
so it sounded not as win to me and I opted to keep it as is. on the consumer we'll have to cast the `Iterable<T>`, but that's ok from a TS pov:
<img src="https://github.com/user-attachments/assets/5c9a1873-bb52-4295-8c06-77642955ead8" width="350" />

Reply:
Meh... Got it... Then probably overall it doesn't make a lot of sense to use generics... Thanks for explanation!

Reply:
Your call on whether to keep them or not!

Reply:
I'll keep then at the `Iterable<T>` level